test/latlonbound_test_utilities.py

```diff
@@ -0,0 +1,386 @@
+"""
+This file stores the necessary helper functions for the latlonbox test cases.
+"""
+import copy
+
+import numpy as np
+
+import uxarray.helpers
+
+
+def max_latitude_rad(v1, v2):
+    """Quantitative method to find the maximum latitude between in a great circle arc
+
+    Parameters
+    ----------
+    v1: float array
+        The first endpoint of the great circle arc [lon, lat] in degree east.
+    v2: float array
+        The second endpoint of the great circle arc [lon, lat] in degree east.
+
+    Returns
+    -------
+    float
+        maximum latitude in radian
+    """
+
+    # Find the parametrized equation for the great circle passing through v1 and v2
+    err_tolerance = 1.0e-15
+    b_lonlat = np.deg2rad(v1)
+    c_lonlat = np.deg2rad(v2)
+
+    v1_cart = uxarray.helpers._convert_node_lonlat_rad_to_xyz(np.deg2rad(v1))
+    v2_cart = uxarray.helpers._convert_node_lonlat_rad_to_xyz(np.deg2rad(v2))
+
+    max_section = np.array([v1_cart, v2_cart])  # record the subsection that has the maximum latitude
+
+    # Only stop the iteration when two endpoints are extremely closed
+    while np.absolute(b_lonlat[1] - c_lonlat[1]) >= err_tolerance or np.absolute(
+            b_lonlat[0] - c_lonlat[0]) >= err_tolerance:
+        max_lat = -np.pi  # reset the max_latitude for each while loop
+        v_b = max_section[0]
+        v_c = max_section[1]
+
+        # Divide the angle of v1/v2 into 10 subsections, the leftover will be put in the last one
+        # Update v0 based on max_section[0], since the angle is always from max_section[0] to v0
+        angle_v1_v2_rad = uxarray.helpers.angle_of_2_vectors(v_b, v_c)
+        v0 = np.array(uxarray.helpers._normalize_in_place(np.cross(np.cross(v1_cart, v2_cart), v_b)))
+        avg_angle_rad = angle_v1_v2_rad / 10
+
+        for i in range(0, 10):
+            angle_rad_prev = avg_angle_rad * i
+            if i >= 9:
+                angle_rad_next = angle_v1_v2_rad
+            else:
+                angle_rad_next = angle_rad_prev + avg_angle_rad
```

Comment:
This section could be vectorized, but I'm not entirely clear of the purpose. How many times does this while loop run? Does it ever get stuck? I don't necessarily think it's necessary to vectorize this small of a loop if it's only run a few times.

Reply:
Thank you for your information. This while loop should keep iterating until it converges. With all current input files, the while loop will converge within 10 iterations. 

As I said, this is only a test-case function. And that's why I am unsure about the need to vectorize these functions.